monkey/monkey_island/cc/ui/src/components/ui-components/ChildCheckbox.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import {Button, Form} from 'react-bootstrap';
+
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faCheckSquare} from '@fortawesome/free-solid-svg-icons';
+import {faSquare} from '@fortawesome/free-regular-svg-icons';
+
+import {getComponentHeight} from './utils/HeightCalculator';
+import WarningIcon from './WarningIcon';
+
+function ChildCheckboxContainer(props) {
+  const {
+    enumOptions,
+    id,
+    multiple,
+    required,
+    disabled,
+    autofocus,
+    onPaneClick,
+    onCheckboxClick,
+    selectedValues,
+    isSafe
+  } = props;
+
+  return(
+    <Form.Group
+      style={{height: `${getComponentHeight(enumOptions.length)}px`}}
+      id={id} multiple={multiple} className='choice-block form-control'
+      required={required} disabled={disabled} autoFocus={autofocus}>
+      {
+        enumOptions.map(({value, label}, i) => {
+          return (
+            <ChildCheckbox key={i} onPaneClick={onPaneClick}
+            onClick={onCheckboxClick} value={value}
+            disabled={disabled} label={label} checkboxState={selectedValues.includes(value)}
+            safe={isSafe(value)}/>
+          );
+        }
+      )}
+    </Form.Group>
+  );
+}
+
+function ChildCheckbox(props) {
+  const {
+    onPaneClick,
+    onClick,
+    value,
+    disabled,
+    label,
+    checkboxState,
+    safe
+  } = props;
+
+  let displayLabel = [<span key={'label'} className={'option-text'}>{label}</span>];
+
+  if (!safe) {
+    displayLabel.push(<WarningIcon key="warning-icon"/>)
+  }
+
+  return (
+    <Form.Group onClick={() => onPaneClick(value)}>
+      <Button value={value} variant={'link'} disabled={disabled} onClick={() => onClick(value)}>
```

Comment:
Where and when is the `disabled` prop used?

Reply:
It's a holdover from the original implementation. It gets passed into the AdvancedMultiSelect as a prop and flows down to the master checkbox, as well as the child checkboxes.

I don't know if any parent components ever actually disable an AdvancedMultiSelect.

Reply:
If we don't support it, we should probably ignore it instead of passing it down to child components. But I leave it up to you

Reply:
I never miss an opportunity to remove code :)